### ISSUE TITLE
Move platform headers to implementation

### ISSUE THREAD

**Issue by edubart** [javalikescript/webview-c#3]
Always including GTK/Windows/Cocoa headers in `webview.h` make some symbols conflicts with some C codebase, also slowdowns compilation of files that includes the header without needing the implementation, also requires the user to install those libraries even if they have a `webview.so` pre compiled. The `webview_priv` could be promoted to a pointer and defined only in the implementation, this way the library could be used with less issues on big C code base, without having the GTK/Windows headers pollute with its symbols.

**Comment by javalikescript** [comment]
Hello,
Thank you for the help and feedback on the library.
I would like to better understand this issue: in my understanding the platform headers are only included depending on the defined target: WEBVIEW_GTK, WEBVIEW_WINAPI or WEBVIEW_COCOA. I do not install GTK on Windows and vice versa. Did I missed something?
Additionally, I would prefer not to use multiple allocations if it is not necessary.
Have a good day,
Samuel

**Comment by edubart** [comment]
This is an enhancement to make the header portable when it's used without the implementation, say you want to use GTK version with the implementation precompiled to `webview.so`, without this the user need to always have  `<JavaScriptCore/JavaScript.h>` , `<gtk/gtk.h>` and `<webkit2/webkit2.h>` headers in the system, and to include the headers he needs to always use `pkg-config --cflags` adding complexity in the build system. 

With this we can precompile the implementation to a `webview.so` or a `webview.dll`, and then distribute only `webview.h` with `webview.so`, without asking the user to install GTK development packages or setup `pkg-config` in his build system.
Also with this when using the webview interface will put less work on the C compiler due to less headers included so it compiles faster. Also some symbols like `id` from Cocoa are problematic, if the user defines a variable with name `id` for example the code won't compile, so with this there are less symbol collisions.

This is a usability enhancement.

Anyway if you don't want to this due to the pointers, I'll keep using my fork, I need this to make this library easy and fast to use on the language I am working on. I made some demos using this library on my language already at [nelua-webview-demo](https://github.com/edubart/nelua-webview-demo). But I've made some changes in the library to make it more portable, including this one, so my users can use with it ease.
